src/ZkFold/Symbolic/Data/Maybe.hs

```diff
@@ -48,3 +50,22 @@ instance SymbolicData a (u (ArithmeticCircuit a))
     restore (h:ts) = Maybe h (restore ts)
     restore _      = Haskell.error "restore ArithmeticCircuit: wrong number of arguments"
     typeSize = 1 + typeSize @a @(u (ArithmeticCircuit a))
+
+maybe :: forall a b f .
+    Conditional (Bool a) b =>
+    DiscreteField (Bool a) a =>
+    b -> (f a -> b) -> Maybe f a -> b
+maybe d h x@(Maybe _ v) = bool @(Bool a) d (h v) $ isNothing x
+
+filter ::
+    Conditional (Bool a) [f a] =>
```

Comment:
This constraint cannot be satisfied in arithmetic circuits

Reply:
@TurtlePU Haven't found alternative solution yet.

Reply:
Your previous solution by folding with maybe as a result was okay though

Reply:
Ok, let's not do `filter` in this PR. It is not quite clear what the result should be.